modules/apps/adaptive-media/adaptive-media-image-content-transformer-backwards-compatibility/src/main/java/com/liferay/adaptive/media/image/content/transformer/backwards/compatibility/internal/AMBackwardsCompatibilityHtmlContentTransformer.java

```diff
@@ -143,7 +144,9 @@ protected Pattern getPattern() {
 	protected String getReplacement(String originalImgTag, FileEntry fileEntry)
 		throws PortalException {
 
-		if (fileEntry == null) {
+		if ((fileEntry == null) ||
+			!ImageProcessorUtil.isSupported(fileEntry.getMimeType())) {
```

Comment:
This is adaptive media; you should use `AMImageMimeTypeProvider` to validate the mime types.  The set of mime types supported by adaptive media are not necessarily the same as those of the image processor.

Reply:
See also `AMImageValidator.isProcessingSupported()`. That's probably what you want to use here.

Reply:
Thanks @adolfopa for your feedback, I updated the code. Please help take a look again. Thanks.

Reply:
`AMImageValidator.isProcessingSupported()` will return true for non-image files so the issue still occurs. So I use `AMImageMimeTypeProvider.isMimeTypeSupported` instead.